.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,50 @@
+name: Pre-commits
+
+on: [push]
+
+jobs:
+  checks:
+    name: Run pre-commit
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Environment
+        run: |
+          echo "NODE_VERSION=18.18.0" >> $GITHUB_ENV
+          echo "PYTHON_VERSION=$(cat .tool-versions | grep -Po '(?<=python ).*')" >> $GITHUB_ENV
+          if [ $GITHUB_REF == 'refs/heads/master' ]; then echo "ENV=prod"; else echo "ENV=staging"; fi >> $GITHUB_ENV
+          echo "REPOSITORY_NAME=${GITHUB_REPOSITORY#*\/}" >> $GITHUB_ENV
+
+      - name: Install NPM dependencies
+        run: npm install
+
+      - name: Install Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      - name: Store python version details for cache
+        run: echo "PY=$(python -VV | sha256sum | cut -d' ' -f1)" >> "$GITHUB_ENV"
+
+      - name: Pre-commit installation cache
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/pre-commit
+          key: pre-commit|${{ env.PY }}|${{ hashFiles('.pre-commit-config.yaml') }}
+          restore-keys: |
+            pre-commit|${{ env.PY }}|
+
+      - name: Install pre-commit
+        run: |
+          python -m pip install pre-commit
```

Comment:
We can remove python and npm dependencies, only Ruby is needed

Reply:
Python needed for pre-commit install - npm yes